libs/product/driver/shopify/testing/src/factories/product-node.factory.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { faker } from '@faker-js/faker/locale/en_US';
+
+import { DaffModelFactory } from '@daffodil/core/testing';
+import {
+  ShopifyProductNode,
+  shopifyHandleTransformer,
+} from '@daffodil/product/driver/shopify';
+
+class MockShopifyProductNode implements ShopifyProductNode {
+  title = faker.commerce.productName();
+  handle = shopifyHandleTransformer(faker.commerce.productName());
+  id = `gid://shopify/Product/${faker.datatype.number({ min: 100000000000 })}`;
+  description = faker.commerce.productDescription();
+  onlineStoreUrl = faker.internet.domainName();
+  availableForSale = faker.datatype.boolean();
+  priceRange = {
+    maxVariantPrice: {
+      amount: parseFloat(faker.commerce.price()),
+      currencyCode: faker.finance.currencyCode(),
+    },
+  };
+  images = {
+    nodes: [
+      {
+        id: `gid://shopify/ProductImage/${faker.datatype.number({ min: 10000000000 })}`,
+        url: faker.image.imageUrl(),
+        altText: faker.random.words(5),
+      },
+    ],
```

Comment:
Use `ShopifyProductImagesFactory.createMany` to generate these. See `DaffCategoryFactory` for an example of how to use child factories.